Description: Fix -w option to leave an existing unique MBR signature intact
 This makes the implementation match the documentation.
Author: Colin Watson <cjwatson@ubuntu.com>
Bug-Ubuntu: https://bugs.launchpad.net/bugs/1065261
Forwarded: no
Last-Update: 2013-02-01

Index: b/src/lib/disk.c
===================================================================
--- a/src/lib/disk.c
+++ b/src/lib/disk.c
@@ -352,7 +352,7 @@
 		printf("******************************************************\n\n");
 		
 	}
-	else if (opts.write_signature) {
+	else if (!mbr->unique_mbr_signature && opts.write_signature) {
 		
 		/* MBR Signatures must be unique for the 
 		   EFI Boot Manager
